### PR TITLE
Only deserialize the docId on the initial deserialize of the header.

### PR DESCRIPTION
**Pull request by baldersheim**
Just skip and verify on subsequent invocations.
@vekterli or @toregge PR

### CODE REVIEW

File: document/src/vespa/document/update/documentupdate.cpp [hunk 1]
```diff
@@ -22,8 +22,62 @@ using namespace vespalib::xml;
...
+    }
+    return std::make_pair(type, docId);
+}
+
+const DocumentType *
+deserializeHeader(const DocumentTypeRepo &repo, vespalib::nbostream & stream, vespalib::stringref & documentId)
+{
+    size_t sz = strnlen(stream.peek(), stream.size());
+    documentId = vespalib::stringref(stream.peek(), sz);
+    stream.adjustReadPos(sz + 1);
```
[the remark this answers is not in the document]
Comment (toregge) [reply document/src/vespa/document/update/documentupdate.cpp]: This code also looks to be very similar to readCStr(), use a call to readCStr() instead ?

**Review by toregge — APPROVED** [review]
LGTM

**Review by vekterli — APPROVED** [review]
:+1:

File: document/src/vespa/document/update/documentupdate.cpp [hunk 2]
```diff
@@ -22,8 +22,62 @@ using namespace vespalib::xml;
...
+
+constexpr unsigned char CONTENT_HASTYPE = 0x01;
+
+vespalib::stringref
+readCStr(nbostream & stream) {
+    const char * s = stream.peek();
+    size_t sz = strnlen(s, stream.size());
+    stream.adjustReadPos(sz+1);
+    return vespalib::stringref(s, sz);
+}
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply document/src/vespa/document/update/documentupdate.cpp]: Will do that separately

File: document/src/vespa/document/update/documentupdate.cpp [hunk 3]
```diff
@@ -22,8 +22,62 @@ using namespace vespalib::xml;
...
-static const unsigned char CONTENT_HASTYPE = 0x01;
+namespace {
+
+constexpr unsigned char CONTENT_HASTYPE = 0x01;
+
+vespalib::stringref
+readCStr(nbostream & stream) {
+    const char * s = stream.peek();
+    size_t sz = strnlen(s, stream.size());
+    stream.adjustReadPos(sz+1);
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply document/src/vespa/document/update/documentupdate.cpp]: Correct, but adjustReadPos will fail the stream then.

**Pull request merged**